tests/component.js

```diff
@@ -0,0 +1,351 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Vue from 'vue';
+import { mount } from '@vue/test-utils';
+import CKEditorComponent from '../src/ckeditor';
+import sinon from 'sinon';
+import { getEditorNamespace } from '../src/utils/geteditornamespace';
+
+/* global window */
+const CKEDITOR = window.CKEDITOR;
+
+describe( 'CKEditor Component', () => {
+	const spies = {};
+
+	let skipReady = false;
+	let sandbox, wrapper, component, props;
+
+	beforeEach( done => {
+		spies.replace = sinon.spy( CKEDITOR, 'replace' );
+		spies.inline = sinon.spy( CKEDITOR, 'inline' );
+
+		wrapper = createComponent( props );
+		component = wrapper.vm;
+		sandbox = sinon.createSandbox();
+
+		if ( skipReady ) {
+			done();
+		} else {
+			component.$once( 'ready', () => {
+				done();
+			} );
+		}
+	} );
+
+	afterEach( () => {
+		skipReady = false;
+
+		for ( const key in spies ) {
+			spies[ key ].restore();
+		}
+
+		wrapper.destroy();
+		sandbox.restore();
+	} );
+
+	describe( 'initialization', () => {
+		it( 'component should have a name', () => {
+			expect( CKEditorComponent.name ).to.equal( 'ckeditor' );
+		} );
+
+		it( 'should render', () => {
+			expect( wrapper.html() ).to.not.be.empty;
+		} );
+
+		describe( 'property', () => {
+			[ {
+				property: 'value',
+				defaultValue: ''
+			}, {
+				property: 'type',
+				defaultValue: 'classic',
+			}, {
+				property: 'editorUrl',
+				defaultValueRegex: /https:\/\/cdn\.ckeditor\.com\/4\.\d{1,2}\.\d{1,2}\/(standard|basic|full)(-all)?\/ckeditor\.js/
+			}, {
+				property: 'config',
+				defaultValue: undefined
+			}, {
+				property: 'tagName',
+				defaultValue: 'textarea'
+			}, {
+				property: 'readOnly',
+				defaultValue: null
+			} ].forEach( ( { property, defaultValue, defaultValueRegex } ) => {
+				it( `"${ property }" should have default value`, () => {
+					if ( defaultValue ) {
+						expect( component[ property ] ).to.equal( defaultValue );
+					}
+
+					if ( defaultValueRegex ) {
+						expect( component[ property ] ).to.match( defaultValueRegex );
+					}
```

Comment:
Personally I'd simplify it by accepting only regexes. This way string values would become `/^string$/`.